crates/bevy_ecs/src/component.rs

```diff
@@ -1622,11 +1641,13 @@ impl<T: Component> FromWorld for InitComponentId<T> {
     }
 }
 
-/// An error returned when the registration of a required component fails.
+/// An error returned when the registration of a related component fails.
+///
+/// See [`RelatedComponents`] for the place where this information is stored and [`Relatedness`] for the strictness of the relationship.
 #[derive(Error, Display, Debug)]
 #[non_exhaustive]
-pub enum RequiredComponentsError {
-    /// The component is already a directly required component for the requiree.
+pub enum RelatedComponentsError {
+    /// The component is already a directly related component for the requiree.
```

Comment:
Hrm, what to do about `requiree`?

I wonder if we should just come up with some convention for referring to each end of a related component, in the vein of "source" (requiree) and "destination" (required), or "subject" and "related", or (insert other bikeshedding ideas).